apm-agent-core/src/main/java/co/elastic/apm/agent/impl/circuitbreaker/CircuitBreakerConfiguration.java

```diff
@@ -0,0 +1,67 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2020 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.impl.circuitbreaker;
+
+import co.elastic.apm.agent.configuration.converter.TimeDuration;
+import co.elastic.apm.agent.configuration.converter.TimeDurationValueConverter;
+import org.stagemonitor.configuration.ConfigurationOption;
+import org.stagemonitor.configuration.ConfigurationOptionProvider;
+
+import static co.elastic.apm.agent.configuration.validation.RangeValidator.isNotInRange;
+
+public class CircuitBreakerConfiguration extends ConfigurationOptionProvider {
+    public static final String CIRCUIT_BREAKER_CATEGORY = "Circuit-Breaker";
+
+    private final ConfigurationOption<Boolean> circuitBreakerEnabled = ConfigurationOption.booleanOption()
+        .key("circuit_breaker_enabled")
+        .tags("added[1.14.0]")
+        .configurationCategory(CIRCUIT_BREAKER_CATEGORY)
+        .description("A boolean specifying whether the circuit breaker should be enabled or not. \n" +
+            "When enabled, the agent periodically polls stress monitors to detect system/process/JVM stress state. \n" +
+            "If ANY of the monitors detects a stress indication, the agent will become inactive, as if the \n" +
+            "<<config-active>> configuration option has been set to `false`, thus reducing resource consumption to a minimum. \n" +
+            "When inactive, the agent continues polling the same monitors in order to detect whether the stress state \n" +
+            "has been relieved. If ALL monitors approve that the system/process/JVM is not under stress anymore, the \n" +
+            "agent will resume and become fully functional.")
+        .dynamic(true)
+        .buildWithDefault(false);
```

Comment:
I think we should enable the circuit breaker by default. If the stress monitors have config options, maybe we don't need this global switch?

Reply:
+1 on keeping it active by default, if config options have "disabled values" that make equivalent behavior that will be one less parameter to document and to explain.

Reply:
We are thinking of making it on by default in the near future, in which case, in order to turn it off, users will have to change multiple configurations if we do that. 
I am thinking about someone getting sporadic pauses with their agents because of some specific JVM or other environmental setups, being required to do that. Moreover, I made sure it is dynamic as well so users can turn off while it's running, without being required to figure out which monitor settings need to be changed.
I already documented and explained, so currently the additional work is to remove.